packages/wdio-browserstack-service/src/cli/index.ts

```diff
@@ -0,0 +1,412 @@
+                    if (line.toLowerCase().includes('ready')) {
+                        this.loadCliParams(cliOut)
+                        PerformanceTester.end(PerformanceEvents.SDK_CLI_START)
+                        resolve()
+                        return
+                    }
+                }
+            })
+
+            this.process!.stderr!.on('data', (data: Buffer) => {
+                this.logger.error(`CLI stderr: ${data.toString().trim()}`)
+            })
+
+            this.process!.on('error', (err: Error) => {
+                cliOut.error = `Error in start: ${err.message}`
+                PerformanceTester.end(PerformanceEvents.SDK_CLI_START, false, err)
+                reject(new Error(`Failed to start CLI process: ${err.message}`))
+            })
+
+            this.process!.on('close', (code: number) => {
+                if (code !== 0) {
+                    reject(new Error(`CLI process exited with code ${code}`))
+                }
+            })
+        })
+    }
+
+    /**
+     * Stop the CLI
+     * @returns {Promise<void>}
+     */
+    async stop() {
+        PerformanceTester.start(PerformanceEvents.SDK_CLI_ON_STOP)
+        this.logger.debug('stop: CLI stop triggered')
+        try {
+
+            await this.unConfigureModules()
+
+            //   grpc channel close
+            //   if (this.channel) {
+            //     // In Node.js, we would use something like:
+            //     await new Promise((resolve) => {
+            //       this.channel.close(() => {
+            //         this.logger.info('stop: channel closed');
+            //         resolve();
+            //       });
+            //     });
+            //     // Allow 5 seconds for shutdown like in Java
+            //     await new Promise(resolve => setTimeout(resolve, 5000));
+            //   }
+
+            if (this.process && this.process.pid) {
+                this.logger.debug('stop: shutting down CLI')
+                this.process.kill()
+
+                // Wait for process to fully exit
+                await new Promise<void>((resolve) => {
+                    let exited = false
+
+                    // Listen for exit event
+                    this.process!.on('exit', () => {
+                        this.logger.debug('stop: CLI process exited')
+                        exited = true
+                        PerformanceTester.end(PerformanceEvents.SDK_CLI_ON_STOP)
+
+                        resolve()
+                    })
+
+                    // Set a timeout in case process doesn't exit cleanly
+                    setTimeout(() => {
+                        if (!exited) {
+                            this.logger.warn('stop: process exit timeout, forcing kill')
+                            this.process!.kill('SIGKILL')
+                            PerformanceTester.end(PerformanceEvents.SDK_CLI_ON_STOP)
+                            resolve()
+                        }
+                    }, 3000)
```

Comment:
looks like a magic number, lets define it as const on top of the file and give it a meaningful name in upper cases